gc/base/standard/Scavenger.hpp

```diff
@@ -285,22 +287,29 @@ class MM_Scavenger : public MM_Collector
 	MMINLINE uintptr_t getFailedTenureLargestObject() { return _failedTenureLargestObject; };
 	MMINLINE bool failedTenureThresholdReached() { return _failedTenureThresholdReached; };
 
-	void completeScanCache(MM_EnvironmentStandard *env);
-	void incrementalScanCacheBySlot(MM_EnvironmentStandard *env);
+	void completeScanCache(MM_EnvironmentStandard *env, MM_CopyScanCacheStandard* scanCache);
+	void incrementalScanCacheBySlot(MM_EnvironmentStandard *env, MM_CopyScanCacheStandard* scanCache);
 
 	MMINLINE MM_CopyScanCacheStandard *aliasToCopyCache(MM_EnvironmentStandard *env, GC_SlotObject *scannedSlot, MM_CopyScanCacheStandard* scanCache, MM_CopyScanCacheStandard* copyCache);
 	MMINLINE uintptr_t scanCacheDistanceMetric(MM_CopyScanCacheStandard* cache, GC_SlotObject *scanSlot);
 	MMINLINE uintptr_t copyCacheDistanceMetric(MM_CopyScanCacheStandard* cache);
 
 	MMINLINE MM_CopyScanCacheStandard *getNextScanCacheFromList(MM_EnvironmentStandard *env);
-	MMINLINE MM_CopyScanCacheStandard *getSurvivorCopyCache(MM_EnvironmentStandard *env);
-	MMINLINE MM_CopyScanCacheStandard *getDeferredCopyCache(MM_EnvironmentStandard *env);
-
 	void addCopyCachesToFreeList(MM_EnvironmentStandard *env);
-
-	MMINLINE bool isWorkAvailableInCache(MM_CopyScanCacheStandard *scanCache);
 	MMINLINE void addCacheEntryToScanListAndNotify(MM_EnvironmentStandard *env, MM_CopyScanCacheStandard *newCacheEntry);
 
+	MMINLINE bool
+	isWorkAvailableInCache(MM_CopyScanCacheStandard *cache)
+	{
+		return ((NULL != cache) && (cache->scanCurrent < cache->cacheAlloc));
+	}
+
```

Comment:
do you need the null check here too?

Reply:
No good catch.  I missed removing it when I refactored the changes into separate commits for the PR.

Reply:
Fixed in the latest commit